src/main/java/ca/wolfram/beta/symath/parser/AntlrExpressionListener.java

```diff
@@ -0,0 +1,316 @@
+        nodesStack.push(new ArrayList<MathNode>());
+        //System.out.println("Entered Term " + ctx.getText() + " , current stack size: " + nodesStack.size());
+    }
+
+    @Override
+    public void exitTerm(TermContext ctx) {
+        ArrayList<MathNode> children = nodesStack.pop();
+
+        ArrayList<MathNode> multiply = new ArrayList<MathNode>();
+        ArrayList<MathNode> divide = new ArrayList<MathNode>();
+        multiply.add(children.get(0));
+        for (int i = 1; i < children.size(); i++) {
+            if( ctx.getChild(i*2-1).getText().equals("*")) {
+                multiply.add(children.get(i));
+            } else {
+                divide.add(children.get(i));
+            }
+        }
+
+        MathNode term;
+        if (multiply.size() == 1) term = multiply.get(0);
+        else term = MultiplyNode.create(multiply);
+        if (!divide.isEmpty()) {
+            divide.add(0, term);
+            term = DivideNode.create(divide);
+        }
+
+        nodesStack.peek().add(term);
+        //System.out.println("Exit Term, current stack size: " + nodesStack.size());
+    }
+
+    // For a factors in the format "a^b". Created in the format PowerNode(a, b)
+
+    @Override
+    public void enterFactor(FactorContext ctx) {
+        nodesStack.push(new ArrayList<MathNode>());
+        //System.out.println("Entered Factor " + ctx.getText() + " , current stack size: " + nodesStack.size());
+    }
+
+    @Override
+    public void exitFactor(FactorContext ctx) {
+        ArrayList<MathNode> children = nodesStack.pop();
+        MathNode factor = BaseNode.ZERO;
+        if (children.size() == 2) factor = PowerNode.create(children.get(0), children.get(1));
+        else if (children.size() == 1) factor = children.get(0);
+        else System.out.println("FACTOR HAS AN INCORRECT NUMBER OF CHILDREN" + children.size());
+        nodesStack.peek().add(factor);
+        //System.out.println("Exit Factor, current stack size: " + nodesStack.size());
+    }
+
+    // This is just a parser remnant.
+
+    @Override
+    public void enterAtom(AtomContext ctx) {
+        // TODO Auto-generated method stub
+
+    }
+
+    @Override
+    public void exitAtom(AtomContext ctx) {
+        // TODO Auto-generated method stub
+    }
+
+    // for scientific notations in the format "aEb". Created in the format MultiplyNode(a PowerNode(10, b))
+
+    @Override
+    public void enterScientific(ScientificContext ctx) {
+        nodesStack.push(new ArrayList<MathNode>());
+        //System.out.println("Entered Scientific " + ctx.getText() + " , current stack size: " + nodesStack.size());
+    }
+
+    @Override
+    public void exitScientific(ScientificContext ctx) {
+        ArrayList<MathNode> children = nodesStack.pop();
+        MathNode scientific = BaseNode.ZERO;
+        if (children.size() == 2) scientific = MultiplyNode.create(children.get(0), PowerNode.create(BaseNode.create(10l), children.get(1)));
```

Comment:
please use capital L for longs. I believe that is the convention and it avoids confusion with '1'